### PR TITLE
New : pp counter

### PR DESCRIPTION
**Pull request by Inlayo**
Photo : 
Video : https://youtu.be/nyS-b5tggo8
CompatibleWith: all

### CODE REVIEW

File: counters/pp counter by Inlayo/js/index.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
+const HOST = '127.0.0.1:24050';
+const socket = new ReconnectingWebSocket(`ws://${HOST}/ws`);
+let wrapper = document.getElementById('wrapper');
+let ifFcpp = document.getElementsByClassName('ifFcpp')[0];
+let progress = document.getElementById("progress");
```
Comment (cyperdark) [inline counters/pp counter by Inlayo/js/index.js:5]: remove unused selectors

File: counters/pp counter by Inlayo/js/index.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    socket.send("Client Closed!");
+};
+socket.onerror = error => console.log("Socket Error: ", error);
+
+let animation = {
+    pp: new CountUp('pp', 0, 0, 0, 0.5, { decimalPlaces: 2, useEasing: true, useGrouping: false, separator: " ", decimal: "." }),
+    ifFcpp: new CountUp('ifFcpp', 0, 0, 0, 0.5, { decimalPlaces: 2, useEasing: true, useGrouping: false, separator: " ", decimal: "." }),
+    hun: new CountUp('hun', 0, 0, 0, 0.5, { decimalPlaces: 2, useEasing: true, useGrouping: false, separator: " ", decimal: "." }),
+    fiv: new CountUp('fiv', 0, 0, 0, 0.5, { decimalPlaces: 2, useEasing: true, useGrouping: false, separator: " ", decimal: "." }),
+    miss: new CountUp('miss', 0, 0, 0, 0.5, { decimalPlaces: 2, useEasing: true, useGrouping: false, separator: " ", decimal: "." }),
```
Comment (cyperdark) [inline counters/pp counter by Inlayo/js/index.js:19]: and remove unused here as well

File: counters/pp counter by Inlayo/js/index.js [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        tempState = data.menu.state
+    }
+    if (data.gameplay.pp.current !== '' && (tempState === 2 || tempState === 7)) {
+        animation.pp.update(data.gameplay.pp.current)
+    }
+    if (data.menu.pp[100] !== '' && (tempState !== 2 && tempState !== 7)) {
+        animation.pp.update(data.menu.pp[100])
+    }
+    if (data.gameplay.pp.fc !== '' && data.gameplay.pp.fc !== 0) {
+        animation.ifFcpp.update(data.gameplay.pp.fc)
```
Comment (cyperdark) [inline counters/pp counter by Inlayo/js/index.js:41]: here as well

**Review by cyperdark — CHANGES_REQUESTED** [review]

**Review by Inlayo — COMMENTED** [review]
:D

File: counters/pp counter by Inlayo/js/index.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+  }
+  if (data.menu.pp[100] !== "" && tempState !== 2 && tempState !== 7) {
+    animation.pp.update(data.menu.pp[100]);
+  }
+  if (data.gameplay.hits[0] > 0 || data.gameplay.hits.sliderBreaks > 0) {
+    ifFcpp.style.opacity = 1;
+  } else {
+    ifFcpp.style.opacity = 0;
+  }
+  if (fullTime !== data.menu.bm.time.mp3) {
```
Comment (cyperdark) [inline counters/pp counter by Inlayo/js/index.js:44]: unused code

File: counters/pp counter by Inlayo/js/index.js [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+  if (data.menu.state !== tempState) {
+    tempState = data.menu.state;
+  }
+  if (data.gameplay.pp.current !== "" && (tempState === 2 || tempState === 7)) {
+    animation.pp.update(data.gameplay.pp.current);
+  }
+  if (data.menu.pp[100] !== "" && tempState !== 2 && tempState !== 7) {
+    animation.pp.update(data.menu.pp[100]);
+  }
+  if (data.gameplay.hits[0] > 0 || data.gameplay.hits.sliderBreaks > 0) {
```
Comment (cyperdark) [inline counters/pp counter by Inlayo/js/index.js:39]: unused code

File: counters/pp counter by Inlayo/js/index.js [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+    ifFcpp.style.opacity = 1;
+  } else {
+    ifFcpp.style.opacity = 0;
+  }
+  if (fullTime !== data.menu.bm.time.mp3) {
+    fullTime = data.menu.bm.time.mp3;
+    onepart = 178 / fullTime;
+  }
+  if (
```
Comment (cyperdark) [inline counters/pp counter by Inlayo/js/index.js:48]: unused code

**Review by cyperdark — CHANGES_REQUESTED** [review]

**Review by Inlayo — COMMENTED** [review]
:D

**Pull request merged**